### PR TITLE
added readmes with new sample locations

### PR DESCRIPTION
**Pull request by patilsnr**

### CODE REVIEW

File: provisioning/Samples/service/GroupCertificateVerificationSample/GroupCertificateVerificationSample.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (patilsnr) [inline provisioning/Samples/service/GroupCertificateVerificationSample/GroupCertificateVerificationSample.sln:2]: For whatever reason, this .sln file didn't seem to exist in the samples repo before so my machine automatically added it.

File: iot-hub/Samples/device/DeviceReconnectionSample/readme.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Device Reconnection Sample
 
-This sample code demonstrates the various connection status changes and connection status change reasons the device client can return, and how to handle them.
+
```
Comment (drwill-ms) [inline iot-hub/Samples/device/DeviceReconnectionSample/readme.md:3]: nit: extra newline

File: iot-hub/Samples/device/PnpDeviceSamples/readme.md [hunk 3]
```diff
@@ -14,6 +14,11 @@ urlFragment: azure-iot-pnp-device-samples-for-csharp-net
 
 # IoT Plug And Play device samples
 
+The samples that previously existed here have been removed since this sample repository will be retired and archived.
+
+Please see here for the up to date [PnpDeviceSamples](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples/solutions/PnpDeviceSamples).
+
+
 [![Documentation](../../../../docs/media/docs-link-buttons/azure-documentation.svg)](https://docs.microsoft.com/azure/iot-develop/)
```
Comment (drwill-ms) [inline iot-hub/Samples/device/PnpDeviceSamples/readme.md:22]: You can remove everything else. We don't want to risk stuff being out of date.

File: iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md [hunk 4]
```diff
@@ -1,4 +1,7 @@
 # x509 device certificate with chain sample
+The sample that previously existed here has been removed since this sample repository will be retired and archived.
+
+Please see here for the up to date [X509Sample](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples/how%20to%20guides/X509DeviceCertWithChainSample).
 
 ## How to run the sample
```
Comment (drwill-ms) [inline iot-hub/Samples/device/X509DeviceCertWithChainSample/readme.md:6]: Same everywhere.

File: provisioning/Samples/service/GroupCertificateVerificationSample/GroupCertificateVerificationSample.sln (hunk 1 above)
Comment (drwill-ms) [reply provisioning/Samples/service/GroupCertificateVerificationSample/GroupCertificateVerificationSample.sln:2]: You can delete it. We want to remove all the samples too.

**Review by drwill-ms — CHANGES_REQUESTED** [review]
Remove everything from the readmes but the "sample has moved" messages.

When will you remove the actual samples?

File: iot-hub/Samples/device/TwinSample/readme.md [hunk 5]
```diff
@@ -1,14 +1,5 @@
...
-This sample code demonstrates device twins support with IoT hub.
-
-For more info about device twins:  
-- [Device Twin getting started](https://docs.microsoft.com/azure/iot-hub/iot-hub-csharp-csharp-twin-getstarted)
-- [Device Twins Developer Guide](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-device-twins)
-
-> Note for Plug and Play (PnP) device twins support with IoT hub and IoT Central, there are specific conventions of the JSON document to follow:
-> - [PnP device developer guide](https://docs.microsoft.com/azure/iot-develop/concepts-developer-guide-device?pivots=programming-language-csharp)
-> - [PnP device samples](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/PnpDeviceSamples)
+Please see here for the up to date [TwinSample](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples/getting%20started/TwinSample).
```
Comment (drwill-ms) [inline iot-hub/Samples/device/TwinSample/readme.md:5]: Not the right sample link, I think.

File: iot-hub/Samples/module/ModuleSample/readme.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 # Module Sample
 
-This sample illustrates how a module client handles its connection status updates, while sending telemetry to IoT Hub, and sending and receiving twin property updates.
+The sample that previously existed here has been removed since this sample repository will be retired and archived.
+
+Please see here for the up to date [ModuleSample](https://github.com/Azure/azure-iot-sdk-csharp/tree/main/iothub/device/samples).
```
Comment (drwill-ms) [inline iot-hub/Samples/module/ModuleSample/readme.md:5]: I don't see a module sample here (and it is missing the next folder).

Comment (drwill-ms) [reply iot-hub/Samples/module/ModuleSample/readme.md:5]: e.g. getting started

**Pull request merged**